### PR TITLE
Azure data explorer client credentials

### PR DESCRIPTION
**Pull request by noydavidi**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-10638)

## Description
Added support for the *Client Credentials* authentication flow.

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md [hunk 1]
```diff
@@ -2,18 +2,35 @@ In order to connect to the Azure Data Explorer using either Cortex XSOAR Azure A
...
+3. *Client Credentials Flow*
+
+## Self-Deployed Azure App
+
+To use a self-configured Azure application, you need to add a new Azure App Registration in the Azure Portal.
+To add the registration, refer to the following [Microsoft article](https://learn.microsoft.com/en-us/defender-xdr/api-create-app-web?view=o365-worldwide) steps 1-8.
+
+### Required permissions
+
+- Azure Service Management - permission `user_impersonation` of type Delegated
```
Comment (michal-dagan) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md:14]: Azure Data Explorer or Azure Service Management?

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md [hunk 2]
```diff
@@ -28,15 +45,25 @@ to link Azure Data Explorer with Cortex XSOAR.
 
 At end of the process you'll see a message that you've logged in successfully.
 
-### Cortex XSOAR Azure App
```
Comment (michal-dagan) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md:31]: please keep it under the `Device Code Flow` section:)
but maybe as a section under it with ####

File: Packs/AzureDataExplorer/ReleaseNotes/1_3_0.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Azure Data Explorer
+
+- Added a Client Credentials flow.
```
Comment (michal-dagan) [inline Packs/AzureDataExplorer/ReleaseNotes/1_3_0.md:6]:
```suggestion
Added support for the *Client Credentials* authentication flow.
```

**Review by JudahSchwartz — APPROVED** [review]
amazing! 🧞

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md [hunk 4]
```diff
@@ -1,6 +1,72 @@
 Use the Azure Data Explorer integration to collect and analyze data inside Azure Data Explorer clusters, and to manage search queries.
 This integration was integrated and tested with version V1 of AzureDataExplorer.
 
+In order to connect to the Azure Data Explorer using either Cortex XSOAR Azure App or the Self-Deployed Azure App, use one of the following methods:
```
Comment (michal-dagan) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md:4]:
```suggestion
# Authorization

In order to connect to the Azure Data Explorer using either Cortex XSOAR Azure App or the Self-Deployed Azure App, use one of the following methods:
```

**Review by michal-dagan — APPROVED** [review]
Great!

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer.yml [hunk 5]
```diff
@@ -38,26 +38,27 @@ configuration:
   required: true
   defaultvalue: Device Code
   type: 15
-  additionalinfo: Type of authentication - could be Authorization Code Flow (recommended) or Device Code Flow
+  additionalinfo: Type of authentication - could be Authorization Code Flow (recommended), Device Code Flow or Client Credentials Flow
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer.yml:41]:
```suggestion
  additionalinfo: Type of authentication - could be Authorization Code Flow (recommended), Device Code Flow, or Client Credentials Flow.
```

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md [hunk 6]
```diff
@@ -2,18 +2,35 @@ In order to connect to the Azure Data Explorer using either Cortex XSOAR Azure A
 
 1. *Authorization Code Flow* (Recommended).
 2. *Device Code Flow*.
+3. *Client Credentials Flow*
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md:5]:
```suggestion
3. *Client Credentials Flow*.
```

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md [hunk 7]
```diff
@@ -2,18 +2,35 @@ In order to connect to the Azure Data Explorer using either Cortex XSOAR Azure A
...
+
+## Self-Deployed Azure App
+
+To use a self-configured Azure application, you need to add a new Azure App Registration in the Azure Portal.
+To add the registration, refer to the following [Microsoft article](https://learn.microsoft.com/en-us/defender-xdr/api-create-app-web?view=o365-worldwide) steps 1-8.
+
+### Required permissions
+
+- Azure Data Explorer - permission `user_impersonation` of type Delegated
+- Microsoft Graph - permission `offline_access` of type Delegated
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md:15]:
```suggestion
- Azure Data Explorer - permission `user_impersonation` of type Delegated.
- Microsoft Graph - permission `offline_access` of type Delegated.
```

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md [hunk 8]
```diff
@@ -2,18 +2,35 @@ In order to connect to the Azure Data Explorer using either Cortex XSOAR Azure A
...
+
+To add a permission:
+
+1. Navigate to **Home** > **App registrations**.
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md:19]: Is this in the Azure portal?  Please specify.

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md [hunk 9]
```diff
@@ -2,18 +2,35 @@ In order to connect to the Azure Data Explorer using either Cortex XSOAR Azure A
...
+2. Search for your app under 'all applications'.
+3. Click **API permissions** > **Add permission**.
+4.  Search for the specific Microsoft API and select the specific permission of type Delegated.
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md:22]:
```suggestion
4. Search for the specific Microsoft API and select the specific permission of type Delegated.
```

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md [hunk 10]
```diff
@@ -2,18 +2,35 @@ In order to connect to the Azure Data Explorer using either Cortex XSOAR Azure A
...
-1. To use a self-configured Azure application, you need to add a new Azure App Registration in the Azure Portal. To add the registration, refer to the following [Microsoft article](https://docs.microsoft.com/en-us/microsoft-365/security/defender/api-create-app-web?view=o365-worldwide#create-an-app) steps 1-8.
-2. In the **Authentication Type** field, select the **Authorization Code** option.
-3. In the **Application ID** field, enter your Client/Application ID. 
-4. In the **Client Secret** field, enter your Client Secret.
-5. In the **Tenant ID** field, enter your Tenant ID .
-6. In the **Application redirect URI** field, enter your Application redirect URI.
-7. Save the instance.
-8. Run the `!azure-data-explorer-generate-login-url` command in the War Room and follow the instruction.
-9. Save the instance.
+1. In the **Authentication Type** field, select the **Authorization Code** option.
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md:26]: Is this in the Azure portal?  Please specify.

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md [hunk 11]
```diff
@@ -2,18 +2,35 @@ In order to connect to the Azure Data Explorer using either Cortex XSOAR Azure A
...
+2. In the **Application ID** field, enter your Client/Application ID. 
+3. In the **Client Secret** field, enter your Client Secret.
+4. In the **Tenant ID** field, enter your Tenant ID .
+5. In the **Application redirect URI** field, enter your Application redirect URI.
+6. Save the instance.
+7. Run the `!azure-data-explorer-generate-login-url` command in the War Room and follow the instruction.
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md:32]:
```suggestion
7. Run the `!azure-data-explorer-generate-login-url` command in the War Room and follow the instructions.
```

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md [hunk 12]
```diff
@@ -2,18 +2,35 @@ In order to connect to the Azure Data Explorer using either Cortex XSOAR Azure A
...
-5. In the **Tenant ID** field, enter your Tenant ID .
-6. In the **Application redirect URI** field, enter your Application redirect URI.
-7. Save the instance.
-8. Run the `!azure-data-explorer-generate-login-url` command in the War Room and follow the instruction.
-9. Save the instance.
+1. In the **Authentication Type** field, select the **Authorization Code** option.
+2. In the **Application ID** field, enter your Client/Application ID. 
+3. In the **Client Secret** field, enter your Client Secret.
+4. In the **Tenant ID** field, enter your Tenant ID .
+5. In the **Application redirect URI** field, enter your Application redirect URI.
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/AzureDataExplorer_description.md:30]:
```suggestion
1. In the *Authentication Type* field, select the **Authorization Code** option.
2. In the *Application ID* field, enter your Client/Application ID. 
3. In the *Client Secret* field, enter your Client Secret.
4. In the *Tenant ID* field, enter your Tenant ID .
5. In the *Application redirect URI* field, enter your Application redirect URI.
```

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md [hunk 13]
```diff
@@ -1,6 +1,74 @@
 Use the Azure Data Explorer integration to collect and analyze data inside Azure Data Explorer clusters, and to manage search queries.
 This integration was integrated and tested with version V1 of AzureDataExplorer.
 
+# Authorization
+
+In order to connect to the Azure Data Explorer using either Cortex XSOAR Azure App or the Self-Deployed Azure App, use one of the following methods:
+
+1. *Authorization Code Flow* (Recommended).
+2. *Device Code Flow*.
+3. *Client Credentials Flow*
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md:10]:
```suggestion
3. *Client Credentials Flow*.
```

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md [hunk 14]
```diff
@@ -1,6 +1,74 @@
...
+
+## Self-Deployed Azure App
+
+To use a self-configured Azure application, you need to add a new Azure App Registration in the Azure Portal.
+To add the registration, refer to the following [Microsoft article](https://learn.microsoft.com/en-us/defender-xdr/api-create-app-web?view=o365-worldwide) steps 1-8.
+
+### Required permissions
+
+- Azure Data Explorer - permission `user_impersonation` of type Delegated
+- Microsoft Graph - permission `offline_access` of type Delegated
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md:20]:
```suggestion
- Azure Data Explorer - permission `user_impersonation` of type Delegated.
- Microsoft Graph - permission `offline_access` of type Delegated.
```

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md [hunk 15]
```diff
@@ -1,6 +1,74 @@
...
+3. Click **API permissions** > **Add permission**.
+4.  Search for the specific Microsoft API and select the specific permission of type Delegated.
+
+### Authentication Using the Authorization Code Flow (recommended)
+
+1. In the **Authentication Type** field, select the **Authorization Code** option.
+2. In the **Application ID** field, enter your Client/Application ID. 
+3. In the **Client Secret** field, enter your Client Secret.
+4. In the **Tenant ID** field, enter your Tenant ID .
+5. In the **Application redirect URI** field, enter your Application redirect URI.
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md:35]:
```suggestion
1. In the *Authentication Type* field, select the **Authorization Code** option.
2. In the *Application ID* field, enter your Client/Application ID. 
3. In the *Client Secret* field, enter your Client Secret.
4. In the *Tenant ID* field, enter your Tenant ID .
5. In the *Application redirect URI* field, enter your Application redirect URI.
```

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md [hunk 16]
```diff
@@ -1,6 +1,74 @@
...
+To use a self-configured Azure application, you need to add a new Azure App Registration in the Azure Portal.
+To add the registration, refer to the following [Microsoft article](https://learn.microsoft.com/en-us/defender-xdr/api-create-app-web?view=o365-worldwide) steps 1-8.
+
+### Required permissions
+
+- Azure Data Explorer - permission `user_impersonation` of type Delegated
+- Microsoft Graph - permission `offline_access` of type Delegated
+
+To add a permission:
+
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md:23]: Is this in the Azure portal? Please specify.

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md [hunk 17]
```diff
@@ -1,6 +1,74 @@
...
+1. Navigate to **Home** > **App registrations**.
+2. Search for your app under 'all applications'.
+3. Click **API permissions** > **Add permission**.
+4.  Search for the specific Microsoft API and select the specific permission of type Delegated.
+
+### Authentication Using the Authorization Code Flow (recommended)
+
+1. In the **Authentication Type** field, select the **Authorization Code** option.
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md:31]: Is this in the Azure portal? Please specify.

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md [hunk 18]
```diff
@@ -1,6 +1,74 @@
...
+7. Run the `!azure-data-explorer-generate-login-url` command in the War Room and follow the instruction.
+8. Save the instance.
+   
+### Authentication Using the Device Code Flow
+
+Use the [device code flow](https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication#device-code-flow)
+to link Azure Data Explorer with Cortex XSOAR.
+
+1. Fill in the required parameters.
+2. In the **Authentication Type** field, select the **Device Code** option.
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md:46]:
```suggestion
2. In the *Authentication Type* field, select the **Device Code** option.
```

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md [hunk 19]
```diff
@@ -1,6 +1,74 @@
...
+6. Save the instance.
+7. Run the `!azure-data-explorer-generate-login-url` command in the War Room and follow the instruction.
+8. Save the instance.
+   
+### Authentication Using the Device Code Flow
+
+Use the [device code flow](https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication#device-code-flow)
+to link Azure Data Explorer with Cortex XSOAR.
+
+1. Fill in the required parameters.
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md:45]: Is this in the Azure portal? Please specify.

File: Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md [hunk 20]
```diff
@@ -14,10 +82,10 @@ This integration was integrated and tested with version V1 of AzureDataExplorer.
...
     | Trust any certificate (not secure) |  | False |
     | Use system proxy settings |  | False |
-    | Authentication Type | Type of authentication - could be Authorization Code Flow \(recommended\) or Device Code Flow | False |
-    | Tenant ID (for Authorization Code mode) |  | False |
-    | Client Secret (for Authorization Code mode) |  | False |
-    | Client Secret (for Authorization Code mode) |  | False |
+    | Authentication Type | Type of authentication - could be Authorization Code Flow \(recommended\), Device Code Flow or Client Credentials Flow. | False |
+    | Tenant ID | For Authorization Code or Client Credentials Flows. | False |
+    | Client Secret | For Authorization Code or Client Credentials Flows. | False |
+    | Client Secret | For Authorization Code or Client Credentials Flows. | False |
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/Integrations/AzureDataExplorer/README.md:88]: Why is this twice?

File: Packs/AzureDataExplorer/ReleaseNotes/1_3_0.md [hunk 21]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Azure Data Explorer
+
+- Added support for the *Client Credentials* authentication flow.
```
Comment (ShirleyDenkberg) [inline Packs/AzureDataExplorer/ReleaseNotes/1_3_0.md:6]:
```suggestion
Added support for the *Client Credentials* authentication flow.
```

**Comment by ShirleyDenkberg** [comment]
@JudahSchwartz @michal-dagan Doc review completed.

**Pull request merged**